dbms/src/Storages/DeltaMerge/File/DMFileWriter.cpp

```diff
@@ -128,6 +129,11 @@ void DMFileWriter::addStreams(ColId col_id, DataTypePtr type, bool do_index)
 
 void DMFileWriter::write(const Block & block, const BlockProperty & block_property)
 {
+#ifndef NDEBUG
+    // In the prod env, the #rows is checked in upper level
+    block.checkNumberOfRows();
+#endif
```

Comment:
ditto

Reply:
assert can not be used in here because `block.checkNumberOfRows()` return void